### PR TITLE
Allow selenium image configuration

### PR DESCRIPTION
**Pull request by NoelDeMartin**
I've decided to remove the private properties because they were only used once, and now everything is encapsulated in the `getSeleniumImage` method.

Other than that, there isn't much more to the PR. It allows to configure the selenium image by setting `MOODLE_BEHAT_SELENIUM_IMAGE`, or specify it for each profile with `MOODLE_BEHAT_SELENIUM_CHROME_IMAGE` and `MOODLE_BEHAT_SELENIUM_FIREFOX_IMAGE`.

I also pondered whether to allow indicating only the browser version. For example, setting `MOODLE_BEHAT_CHROME_VERSION` or accepting `chrome:117` for the profile. But I wasn't convinced by that approach, so I ended up using this one that also allows customizing the image completely (for example, in case [you'd need to use seleniarm](https://github.com/moodlehq/moodle-docker/pull/250) or something else).

### CODE REVIEW

**Comment by stronk7** [comment]
I like to be able to specify the images... just guessing when we should follow the `env` route here or just add one option (`--selenium` to the command (that, of course, will take precedence over the profile one).

Right now it's really a mix of env and options, grrr. Need to think a little bit more about it...

**Comment by stronk7** [comment]
OT: @NoelDeMartin, I've force-pushed (without changes) your candidate branch to get the tests executed, it seems that some of them were stuck.

**Comment by stronk7** [comment]
> It seems like some tests are failing in Travis but I cannot see the details.

Yeah, Travis is being a nightmare since 2 weeks ago. Sometimes it installs PostgreSQL on port 5432, others in 5432... I've been fighting with that in #270 , just to, at the end, leave the port like it's now (5432).

Will look to this soon... thanks!

**Comment by stronk7** [comment]
Side note, surely this closes #15

**Comment by stronk7** [comment]
Thanks @NoelDeMartin,

it's 98% perfect for me! The remaining 2% is:

1. Do we really need the 3 env variables? Why isn't the "general" (`MOODLE_BEHAT_SELENIUM_IMAGE`) one enough for everything? Whoever wants to set a custom selenium image just does it using it (or the new option) and done. I've tried to imagine use cases and I'm not sure to get the point about the browser specific ones.
2. No matter the previous point outcome, I think we need to document a little bit more the env variables. Maybe in the changelog, or maybe in the `docs/Help.md` page (there is a Behat section there).

Thoughts?

**Comment by NoelDeMartin** [comment]
Thanks for the review @stronk7, I have applied your suggestions:

1. I just thought it'd made sense to override them individually given that we have 2 different profiles and the same image couldn't work for both. But I don't mind removing them. Most people is likely to use only one profile, and if they don't they can now pass a different `--selenium` flag.
2. There are other env variables such as `MOODLE_START_BEHAT_SERVERS` that are not documented, so I wasn't sure if that was necessary (specially now that we have the documented `--selenium` flag). But I added it in the CHANGELOG.

**Comment by stronk7** [comment]
> There are other env variables such as MOODLE_START_BEHAT_SERVERS that are not documented,

Yeah, I was looking for it in the Docs and only the "quick start" section of https://moodlehq.github.io/moodle-plugin-ci/Help.html was showing some options.

I've created #271 about that, surely we'll need to make a few new pages (similar to the app one) and document many of those environmental variables. In any case that falls out of this PR scope.

Ciao :-)

**Comment by stronk7** [comment]
I will merge this as soon as we are green (tests still running). Thanks for your work and patience!

File: src/Command/BehatCommand.php [hunk 1]
```diff
@@ -65,6 +46,7 @@ protected function configure(): void
             ->addOption('name', null, InputOption::VALUE_REQUIRED, 'Behat name option to use', '')
             ->addOption('start-servers', null, InputOption::VALUE_NONE, 'Start Selenium and PHP servers')
             ->addOption('auto-rerun', null, InputOption::VALUE_REQUIRED, 'Number of times to rerun failures', 2)
+            ->addOption('selenium', null, InputOption::VALUE_NONE, 'Selenium Docker image')
```
Comment (stronk7) [inline src/Command/BehatCommand.php:49]: Oh, sorry for not detecting this earlier. I don't think that `InputOption::VALUE_NONE` is correct here. If the option is used, a value (valid selenium image) is required.

Comment (stronk7) [reply src/Command/BehatCommand.php:49]: I'm not sure why the tests are passing, somehow thy are bogus and not validating the `VALUE_NONE` restriction that should avoid that option to be passed with any value.

**Review by stronk7 — CHANGES_REQUESTED** [review]

File: src/Command/BehatCommand.php (hunk 1 above)
Comment (NoelDeMartin) [reply src/Command/BehatCommand.php]: Thanks for spotting that, it seems like I hadn't completely understood how those constants worked. I fixed it now.

Not sure about how to improve the tests, though. But I guess it's not related to this feature so if you think it's important to look into that I'd suggest opening a separate issue.

Comment (stronk7) [reply src/Command/BehatCommand.php]: Yeah, will check why the tester is not catching that `VALUE_NONE` "violation". It should. But it's apart from this PR, 100% agree.

**Comment by stronk7** [comment]
Sold!

**Pull request merged**